implementations/rust/ockam/ockam_api/src/authenticator/direct/client.rs

```diff
@@ -0,0 +1,82 @@
+use miette::IntoDiagnostic;
+use std::collections::{BTreeMap, HashMap};
+
+use ockam::identity::AttributesEntry;
+use ockam::identity::Identifier;
+use ockam_core::api::Request;
+use ockam_core::async_trait;
+use ockam_node::Context;
+
+use crate::authenticator::direct::types::AddMember;
+use crate::cloud::{AuthorityNodeClient, HasSecureClient};
+use crate::nodes::service::default_address::DefaultAddress;
+
+#[async_trait]
+pub trait Members {
+    async fn add_member(
+        &self,
+        ctx: &Context,
+        identifier: Identifier,
+        attributes: BTreeMap<String, String>,
+    ) -> miette::Result<()>;
+
+    async fn delete_member(&self, ctx: &Context, identifier: Identifier) -> miette::Result<()>;
+
+    async fn list_member_ids(&self, ctx: &Context) -> miette::Result<Vec<Identifier>>;
+
+    async fn list_members(
+        &self,
+        ctx: &Context,
+    ) -> miette::Result<HashMap<Identifier, AttributesEntry>>;
+}
+
+#[async_trait]
+impl Members for AuthorityNodeClient {
+    async fn add_member(
+        &self,
+        ctx: &Context,
+        identifier: Identifier,
+        attributes: BTreeMap<String, String>,
+    ) -> miette::Result<()> {
+        let req = Request::post("/").body(AddMember::new(identifier).with_attributes(attributes));
+        self.get_secure_client()
+            .tell(ctx, DefaultAddress::DIRECT_AUTHENTICATOR, req)
+            .await
+            .into_diagnostic()?
```

Comment:
Maybe we should implement some syntactic sugar to stop this `.into_diagnostic().success().into_diagnostic()` dance?

Reply:
The easy way to do that would be to add miette's `Diagnostic` derive to our errors or implement some `From<miette::?>`, but we probably don't want to do that. We could also not use miette in the ockam_api crate... but sounds out of scope.